docs/middlewares.md

```diff
@@ -0,0 +1,38 @@
+# Middlewares
+
+Main goal of middleware is to make IaC and remote resources set to match. For this we filter some default resources, mutate or remove some fields in resources or even create and delete new resources.
+
+```go
+type AwsDefaultRoute struct{}
+
+func NewAwsDefaultRoute() AwsDefaultRoute {
+	return AwsDefaultRoute{}
+}
+
+func (m AwsDefaultRoute) Execute(remoteResources, resourcesFromState *[]*resource.Resource) error {
+
+	newRemoteResources := make([]*resource.Resource, 0)
+
+	// ...
+
+	*remoteResources = newRemoteResources
+  
+  return nil
+}
+```
+
+In the above example, we define a middleware called `AwsDefaultRoute` that will modify remote resources. Middleware can access two `*[]*resource.Resource` : the one from IaC side and the other one from remote side and the goal is to rework theses slice to remove false positive drifts. Notice both `remoteResources` and `resourcesFromState` variables are pointers, which mean middlewares can perform mutations on resources before the comparison is made.
```

Comment:
`two *[]*resource.Resource: IaC resources first and then remote resources. The goal ...` (no space before : )
`these slices to ...`